### DISCUSSION TITLE
Where to put #define CIRCULAR_BUFFER_INT_SAFE

### DISCUSSION THREAD

**Discussion by skleeschulte** [rlogiacco/CircularBuffer#22] [Q&A] [answered]
Being new to C/C++, I wonder where I should put `#define CIRCULAR_BUFFER_INT_SAFE`. In [README.md](https://github.com/rlogiacco/CircularBuffer/blob/1.3.1/README.md#interrupts) it is shown before variable declaration:

    #define CIRCULAR_BUFFER_INT_SAFE
    CircularBuffer<unsigned long, 10> timings;

In the [Interrupts.ino example](https://github.com/rlogiacco/CircularBuffer/blob/1.3.1/examples/Interrupts/Interrupts.ino#L1-L3) it is shown before the `#include` statement:

    #define CIRCULAR_BUFFER_INT_SAFE
    #include "CircularBuffer.h"
    CircularBuffer<unsigned long, 10> timings;

Does it matter?

**Comment by proddy** [comment]
think it does, best put it before the #include to be safe

**Comment by rlogiacco** [comment, marked as answer]
@proddy is definitely right, it goes before the import: anywhere after the import and it doesn’t work.
